### DISCUSSION TITLE
Pinouts consistency - name vs gpio # vs schematic

### DISCUSSION THREAD

**Discussion by AlexandreUSA** [Dilbert66/esphome-vistaECP#43] [General]
@Dilbert66 I hope you don't mind these type of comments as they are purely for optimization. I just saw that you changed/documented the pinouts better.  However, I think that you still have some inconsistencies between the comments and the pin numbers. I understand that these pins do not matter but it may be better to have a match between the schematic, the pin names, and their default GPIO numbers. It should be the following in vistaalarm.h to match your schematic:

For ESP8266:
```
#define D1 (5) //scl
#define D2 (4) //sda
#define D5 (14) //sck
```

and for ESP32
```
#define D1 (22) //scl
#define D2 (21) //sda
#define D5 (18) //sck
```

in addition, in vistaalarm.h: you are defining TX, and in vista.h: D6, D7, D8,  and TX. Are these use anywhere?

**Comment by Dilbert66** [comment]
Perhaps you are not aware but in developing code we change things all the time and create many variables that are used at the time.  After some things are changed, we don't always remove existing variables as they tend to get buried with other functions.  They don't hurt anything being there unless the chip is very tight on memory. Those d6,d7, etc are simply defines and do not use any memory so are harmless to have there so I'm not sure why you are concerned about them?  I had some plans for other changes at one time that I have not pursued yet so hence the defines.  

As to the d1,d2,d5 , sure I can change them if it causes confusion.  I added them quickly.

if these d1,d2 are too confusing I can simply have the user configure the pin directly and this eliminates the whole issue.

This way:
```
//pins to use for serial comms to the panel
#ifdef ESP32
#define RX_PIN 22
#define TX_PIN 21
#define MONITOR_PIN 18 // pin used to monitor the green TX line (3.3 level dropped from 12 volts
#else
#define RX_PIN 5
#define TX_PIN 4
#define MONITOR_PIN 14 // pin used to monitor the green TX line (3.3 level dropped from 12 volts
#endif
```

**Reply by AlexandreUSA** [reply]
When I meant optimization, I did not mean it for the code but in small improvement to the documentation. I like your suggestion with the direct definition and even better, let the user defines them in the yaml so no need to touch the rest of the files when an updated code is provided. 

You have done such an amazing work here that I am only looking for small improvements so many more people can use your project.

**Comment by Dilbert66** [comment]
I've pushed a new update to master that fixes a slew of expander issues and adds the changes you recommended ie: configuration of pins via the yaml config file

**Reply by AlexandreUSA** [reply]
Thanks for adding the pin configuration in the Yaml file. 

What was the reason to remove the "armed" state and the RF messages. They were just added last week?

**Comment by Dilbert66** [comment]
geezzzz. I'm working with three different branches and picked up the wrong yaml file to push.  It's corrected now.
